cmd/helm/downloader/chart_downloader.go

```diff
@@ -217,11 +228,11 @@ func VerifyChart(path string, keyring string) (*provenance.Verification, error)
 	return sig.Verify(path, provfile)
 }
 
-// download performs a simple HTTP Get and returns the body.
-func download(href string) (*bytes.Buffer, error) {
+// download performs a HTTP Get using specified client and returns the body.
+func download(href string, client *http.Client) (*bytes.Buffer, error) {
```

Comment:
Based on earlier comments, this would take a second argument that is an interface type:

```
type Getter interface {
  Get(href string) ([]byte, error)
}
```

Then we could pass it either a *ChartRepository or a plain HTTP client (or a simple testing fixture).